sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -78,6 +78,14 @@ class CodegenContext {
   var currentVars: Seq[ExprCode] = null
 
   /**
+   * Whether should we copy the result rows or not.
+   *
+   * If any operator inside WholeStageCodegen generate multiple rows from a single row (for
+   * example, Join), this should be true.
+   */
```

Comment:
I think it's worth nothing that operators which start new pipelines should reset this to false.